Merge master to 1.3 branch

phpipam opened the pull request.

Commit:
Added PhpStorm/Jetbrains project folder to .gitignore

Commit:
top10 hosts widget: removed duplicated code

Commit:
pstn-prefixes: removed some duplicated code

Commit:
menu section: again removed some duplicated code

GaryAllan committed:
Update changelog to work with new class.Rewrite model. #41

GaryAllan committed:
Bugfix: Rework patch for XSS (reflected) in SubnetID param. #1702

phpipam committed:
Update PEAR DNS2 package to latest version

phpipam committed:
PEAR/DNS2 php72 each compatibility

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Changelog update

GaryAllan committed:
Bugfix: Fix scrutinizer critical warnings

GaryAllan committed:
Bugfix: Ensure users are authenticated when calling app/dashboard/widgets/top10_hosts_lib.php

fwdIT committed:
Fix due to 'Change dns_name to hostname'

GaryAllan committed:
Merge pull request #1722 from fwdIT/hostname_field_correction

Fix due to 'Change dns_name to hostname', #1722

GaryAllan committed:
Additional fixes due to 'Change dns_name to hostname', #1722

GaryAllan committed:
Bugfix: Fatal error: Uncaught Error: Class 'Net_DNS2_Resolver' not found. #1718

GaryAllan committed:
Bugfix: Update Net_DNS2_Resolver php72 fixes. #1683, #1718

zombah committed:
fix: Subnet import vrf name check

* It need to be same as vrf import name check

Signed-off-by: Misha Komarovskiy <zombah@gmail.com>

github138 committed:
Update class.Tools.php

Correct encoding for german Umlaute for xls files if necessary

GaryAllan committed:
Merge branch 'github138-patch-1'

GaryAllan committed:
Correct encoding for german Umlaute for xls. #1728

GaryAllan committed:
Merge pull request #1726 from zombah/import-subnet-vrf-fix

fix: Subnet import vrf name check

GaryAllan committed:
Bugfix: notification e-mails contain a broken link. #1708

GaryAllan committed:
Removal of duplicated code

GaryAllan committed:
SubnetsTable, generate settings from internal Tools object

GaryAllan committed:
SubnetsTree, make $subnets_by_id and $children_by_parent_id public

GaryAllan committed:
Bugfix: php5.3 short array fixes. #1730

GaryAllan committed:
Update CHANGELOG

GaryAllan committed:
Provide resized and optimised version of the default built-in
phpIPAM.png logo file for use in email (220x54 52Kb->2.2Kb)

A lossless compression optimised version of the original 640x150
image will be used for $logo_width > default=220 (650x150 52Kb->33Kb)

GaryAllan committed:
pngcrush /css/1.3.1/images/phpipam_logo_smal.png

GaryAllan committed:
Convert Subnets::print_subnet_menu() to consume SubnetsTree class.

GaryAllan committed:
Merge branch 'master' into bootstrap-tables

GaryAllan committed:
Fix merge conflict

Commit:
Use PHP_BINARY instead of PHP_BINDIR to detect the php executable

(cherry picked from commit 56a68bec9aba1da08743e022c0e4ef721748b950)

GaryAllan committed:
Correct PHPDoc comments

GaryAllan committed:
Bugfix: Reflected XSS requiring user interaction on search. #1705

GaryAllan committed:
Bugfix: Code injection in Name field. #1667

GaryAllan committed:
Update CHANGELOG

phpipam committed:
Added BW calc config

phpipam committed:
Merge branch 'master' into bw_calculator

GaryAllan committed:
Use built-in validate_email() common function to verify email address

GaryAllan committed:
Bugfix: Ensure functions/functions.php in included only once.

GaryAllan committed:
Bugfix: Ensure functions/functions.php in included only once. (PowerDNS)

GaryAllan committed:
Bugfix: Ensure CSRF cookie is present when validating. #1703

GaryAllan committed:
Bugfix: CSRF enables open mail relay. #1703

GaryAllan committed:
Merge branch 'master' into bootstrap-tables

GaryAllan committed:
Don't expand children when generating SubnetsMenu

GaryAllan committed:
Subnets->check_permission() performance improvement for non-Admin users

Allow passing of $subnet object if known to avoid database lookup.
Cache results of section/subnet permission checks.

GaryAllan committed:
SubnetsTree performance improvement for non-Admin users

Pass $subnet object to Subnets->check_permission() to avoid a database lookup.

GaryAllan committed:
Performance Improvement

Avoid caching the large result set of fetch_section_subnets() where it
provides no benefit (no consumers of the cache before the script terminates).

ixdaviddpd committed:
Allow dots to be used in domain names.

GaryAllan committed:
Bugfix: XSS reflected via CSRF (preview.php). #1706

avkolbasenko committed:
Update print-address-table.php

Close #1621 issue

ixdaviddpd committed:
The bug here that is fixed is small, for some reason, something was
changing the periods in vlan domains to underscores.  I can't seem
to find what is doing that.  So fixed that, along with some readablity
of the code (but that could be personal perference).

In English, row and line are synonymous.  The $rowCount varibles is
really the column in the spreadshet, and the Line is the row.  This
was very confusing when trying to read the code.  I suppect that
this syntax is used throughout export-import module.

phpipam committed:
Merge branch 'master' into bw_calculator

phpipam committed:
Merge pull request #1755 from avkolbasenko/master

Update print-address-table.php

phpipam committed:
Fixed #1683 7.2 compatibility

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Fixed required fields error on address deletion

phpipam committed:
Fixed sizeof 7.2 error in class.Log.php

phpipam committed:
Merge pull request #1747 from daviddpd/ixdpd2018

Allow dots to be used in domain names.

phpipam committed:
Merge pull request #1733 from cyberjacob/fix-binary-detection

Use PHP_BINARY instead of PHP_BINDIR to detect the php executable

phpipam committed:
Dropping php 5.3 support with optional override in config.php #1732

phpipam committed:
Typo fix

GaryAllan committed:
Subnets->check_permission() performance improvement for non-Admin users

Ensure supplied $subnet object is valid and contains required properties.

GaryAllan committed:
Widgets top10. Calculate correct length of descriptions containing multi-byte characters. #1753

GaryAllan committed:
Bugfix: Use multi-byte compatible functions when shortening text. #1753

ixdaviddpd committed:
In English, row and line are synonymous. This changes all references of
lineCount to curRow and rowCount to curColumn

ixdaviddpd committed:
L2 domain exporting seemed to have already been implemented, so enable
it via the javascript file.

GaryAllan committed:
top10_percentage widget

- Deduplicate shared code into top10_hosts_lib.php.
- Cleanup of top10_hosts_lib.php code.
- Use multi-byte compatible functions when shortening text. #1753

phpipam committed:
Update jQueryUI

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Merge branch 'master' into bw_calculator

phpipam committed:
BW calculator

phpipam committed:
changed IPcalc title

phpipam committed:
BW calculator widget

GaryAllan committed:
Bugfix: Scan subnet reports invalid php path. #1757

daviddpd committed:
Enbale L2 Domain import.

Fix all templates so that header fields in the template so that all
the template fields provided are automatically matched.

This needed to change the java script, as this matching is done client side,
removing a string replacement of HOSTNAME to DNS_NAME.  Doesn't seem to
break any other importing, but not widely tested.

GaryAllan committed:
Update CHANGELOG

GaryAllan committed:
Bugfix: reflected XSS via CSV upload. (#1742)

GaryAllan committed:
Update CHANGELOG

phpipam committed:
Updated calculations and added more delay values

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Merge branch 'master' into bw_calculator

Commit:
Remove user hint upon login.

DisasteR committed:
set app_id when using Crypt Mode API

phpipam committed:
removed some JS code

phpipam committed:
Merge branch 'master' into bw_calculator

GaryAllan committed:
Bugfix: Escape double quotes properly on locations name. #1673

daviddpd committed:
Complete the implementation, in current code style, for
importing and exporting devices and device Types.

daviddpd committed:
Forgot a few L2dom pieces.  Also, allow L2 Domains to load in
the string VLANs and look up the VLANID.

daviddpd committed:
Stray field - was trying to expand rack name.  For now, RackId is needed to link.

phpipam committed:
merge branch 'master' into bw_calculator

phpipam committed:
Added BW calculator to menu config

phpipam committed:
Updated readme.md

phpipam committed:
Merged MASTER changes to bootstrap-tables

phpipam committed:
Table UI changes

phpipam committed:
Fixed card view

phpipam committed:
Removed BDT tables and remapped Admin tables to datatables - others to be fixed, plus buttons not working

phpipam committed:
sorted table fixes

phpipam committed:
Updated tables - prevent sorting

phpipam committed:
php7.2 fixes - sizeof and int

phpipam committed:
UI fixes

phpipam committed:
Fixed popovers and tooltips on table load

GaryAllan committed:
Bugfix: Use multi-byte compatible functions when shortening text. #1753

GaryAllan committed:
PSTN, PHP 7.2 compability #1683

phpipam committed:
Merge pull request #1768 from daviddpd/ixdpd2018-export-import

Ixdpd2018 export import

phpipam committed:
Merge pull request #1766 from DisasteR/issue-1764

set app_id when using Crypt Mode API

phpipam committed:
Merge pull request #1763 from Gmax76/user-authentication-no-hint

Remove user hint upon login.

phpipam committed:
Updated edit user

phpipam committed:
Updated edit groups

phpipam committed:
Updated edit authmethod

phpipam committed:
Merge branch 'bootstrap-tables' of https://github.com/phpipam/phpipam into bootstrap-tables

phpipam committed:
Two sizeof fixes for php72

phpipam committed:
Small JS optimizations

phpipam committed:
Edit nameservers editing

phpipam committed:
pdns JS fixes

phpipam committed:
Removed JS caching

phpipam committed:
settings js

phpipam committed:
JS updates for sections and instructions

phpipam committed:
JS fixes for vlans

phpipam committed:
VRF JS Fizes

phpipam committed:
IP requests JS fixes

phpipam committed:
All sections UI fix

phpipam committed:
UI fixes

phpipam committed:
php72 fixes

phpipam committed:
rollback

phpipam committed:
Rollback fixes

phpipam committed:
Removed SCRIPT_PREFIX prefix from JS and CSS path

Commit:
UI fixes

phpipam committed:
SCRIPT_PREFIX fix

phpipam committed:
Fixed #1784

phpipam committed:
Initial - buggy :)

phpipam committed:
Popup fixes

phpipam committed:
UI fixes

phpipam committed:
UI fixes

phpipam committed:
UI fixes

GaryAllan committed:
UI Fixes

GaryAllan committed:
Move CSRF functions into Crypto class

GaryAllan committed:
Move token generation code into Crypto class

phpipam committed:
UI fixes

phpipam committed:
UI fixes

phpipam committed:
upgrade, update fixes and UI fixes

phpipam committed:
Fixed transparency in usage PIE graph for dark theme

phpipam committed:
Dashboard and dashboard popup fixes

phpipam committed:
UI fixes

phpipam committed:
UI fixes

phpipam committed:
Search dropdown FIX

phpipam committed:
UI fixes

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

Commit:
Default to dark theme is db field is missing

Commit:
UI fixes

phpipam committed:
Minor UI fixes

Commit:
UI fixes

GaryAllan committed:
Move encryption and crypt API functions into Crypto class

Commit:
Fixed subnet editing from slave subnet list

Commit:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Fixed MCRYPT_RIJNDAEL_256 constant error

phpipam committed:
IP address details rack image fix

phpipam committed:
Menu dropdown hover fix

GaryAllan committed:
Bugfix: Fix scrutinizer flagged issues.

GaryAllan committed:
Bugfix: Fix scrutinizer flagged issues.

GaryAllan committed:
Improve Section Subnets search functionality.

Searches in the server-side paginaion code require a CIDR.
Auto-complete partial search string to CIDRs at sensible boundaries. (IPv4 only)

	SEARCH         AUTO-COMPLETED
	10         --> 10.0.0.0/8
	10.100     --> 10.100.0.0/16
	10.100.1   --> 10.100.1.0/24
	10.100.1.8 --> 10.100.1.8/32

phpipam committed:
Fixed #1797

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Fixed #1599

GaryAllan committed:
Add checks and feedback for writeable upload folders.

GaryAllan committed:
Fix address-details.php temporary share address button.

GaryAllan committed:
Recompute fix

GaryAllan committed:
Bugfix: IPv6 subnet in long format (with leading zeros). #1798

Commit:
Fixed vlan number for slave subnets list

Commit:
Merge branch 'master' of https://github.com/phpipam/phpipam

Commit:
Create robots.txt

phpipam committed:
Merge pull request #1818 from benrubson/robots

Create robots.txt

xrobau committed:
Fix a bunch of implicit const to string assumptions

GaryAllan committed:
Bugfix: Bandwidth calculator - 400 Bad Request. #1807

GaryAllan committed:
Bugfix: api uses $this not in object context. #1828

Commit:
Add custom_ prefix to Prefix controller custom fields

shepherdjay committed:
Fix #1545

GaryAllan committed:
Merge pull request #1822 from xrobau/bugfix/fix-const-array

Fix a bunch of implicit const to string assumptions

GaryAllan committed:
Bugfix: Rack module, variable  not defined warning

GaryAllan committed:
Merge pull request #1847 from shepherdjay/master

Fix #1545 rack device start position overwritten

tsaavik committed:
Added legacy 1.0 to 1.1 update

tsaavik committed:
Some tables must be dropped for upgrade to work

melankh committed:
Update phpipam.po

Typo : "Changelo" => "Changelog"

melankh committed:
Merge pull request #1 from melankh/melankh-patch-1

Update phpipam.po

melankh committed:
Update phpipam.po

Fix translation

melankh committed:
Merge pull request #2 from melankh/melankh-patch-1

Update phpipam.po

phpipam committed:
Fixed CSS for table dark theme

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
SMall UI changes

Bbertatum committed:
Add 8U rack image template

Bbertatum committed:
Add 8U to available rack size

GaryAllan committed:
php72 warning fixes.

GaryAllan committed:
Merge pull request #1881 from melankh/master

Fix French translation

GaryAllan committed:
Merge pull request #1891 from Bbertatum/master

Add 8U size rack

GaryAllan committed:
Fix for php53, this allows the php53 is no longer supported message to be displayed

GaryAllan committed:
Bugfix: When calculating overlaps map Subnets with NULL vrfId to the default vrf (0). #1852

avkolbasenko committed:
Change SSL / TLS description (#1791)

Change SSL / TLS description #1791

GaryAllan committed:
Bugfix: Sections with folders returns no subnets using API #1883

GaryAllan committed:
Bugfix: Update #1883 to return folders in result set.

Gmax76 committed:
FIX ISSUE 1883 (#1893)

GaryAllan committed:
Bugfix: Non-Admins missing the button to Add circuit. #1902
Bugfix: php72 warning fixes.

GaryAllan committed:
Bugfix: Reflected xss on 'mac' parameter. #1903

GaryAllan committed:
Bugfix: SNMPv3 fixes. #1628

        Use correct device SNMPv3 details when calling snmp_session->setSecurity().
        Ensure `devices` `snmp_version` is defined as SET(0,1,2,3) for users upgrading from pre 1.3.1 schemas.

Bugfix: php72 warning fixes.

GaryAllan committed:
SNMPv3: Make it clear that the supported AES privacy protocol is AES-128.
        Support for AES-192 & AES-256 modes to be investigated.

mtotone committed:
Bugfix: wrong field name of deviceTypes table when checking device type

GaryAllan committed:
Merge pull request #1836 from devonblzx/prefix-controller-custom-prefix

Add custom_ prefix to Prefix controller custom fields

GaryAllan committed:
Merge pull request #1910 from mtotone/fix-device-tid

Bugfix: wrong field name of deviceTypes table when checking device type #1910

GaryAllan committed:
Bugfix: User role unable to view subnets. #1914

GaryAllan committed:
SNMPv3: v1/v2c community and v3 username values share a common form input field.

GaryAllan committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

GaryAllan committed:
Update CHANGELOG.

GaryAllan committed:
Bugfix: Posible memory leak in app/tools/circuits/all-circuits-map.php #1907

Patch submitted by @Oirbsiu, #1907.

GaryAllan committed:
Bugfix: Firewall Zone/Mapping Bug? #1602, #1719
Bugfix: php72 warnings fixes.

GaryAllan committed:
Bugfix: Append QSA to API call requests (e.g CRYPT api)

GaryAllan committed:
php72 warnings fix

GaryAllan committed:
Bugfix: Crypt API fixes for legacy mcrypt. (php72 PECL php-mcrypt compatible)

phpipam committed:
Removed mcrypt requirement warning on api

GaryAllan committed:
Bugfix: reset-admin-password.php fails on CentOS7 and Ubuntu18.04. #1915

readline_list_history() isn't available if php-readline is built against libedit.
http://php.net/manual/en/function.readline-list-history.php

Check for empty PHPMailer git submodule folder.

GaryAllan committed:
Bugfix: Firewall Zone/Mapping Bug? pt2 #1602, #1719

GaryAllan reviewed (commented):
Hi Miha,

I'm thinking about the `hostname` `dns_name` SCHEMA changes. If we merge as-is; Any users following the 1.3 branch who run a "Verify Database" will create a new blank `hostname` field instead of renaming `dns_name`.

Can we bump master to 1.3.2 before merging to make sure UPDATE.sql runs to minimize reported issues? 

Otherwise big 👍

GaryAllan committed:
Bugfix: Firewall Zone/Mapping Bug? pt3, VLAN names #1602, #1719

GaryAllan committed:
UI color changes for darktheme.

GaryAllan committed:
Fix GitHub download links

GaryAllan committed:
Map to existing NAT error if empty. #1921

GaryAllan committed:
Map to existing NAT error if empty. pt2 #1921

GaryAllan committed:
Bugfix: Warning: Use of undefined constant SETTINGS - assumed 'SETTINGS'
        (this will throw an Error in a future version of PHP)

GaryAllan committed:
Bugfix: Warning: Use of undefined constant PHP_BINARY - assumed 'PHP_BINARY'
        (this will throw an Error in a future version of PHP)

GaryAllan committed:
Update class.Crypto to support mcrypt and OpenSSL encryption modes
to enable easier reuse of the class in API clients.

The default security provider for the Crypt API as of 1.3.2 is now
OpenSSL due to mcrypt removal in php72.

To enable mcrypt backwards compatibility add the following to config.php.

   $api_crypt_encryption_library = "mcrypt";

GaryAllan committed:
Bugfix: Update CSS to generate unique success, warning, error status icon shapes. Red/Green colour bind users. #1936

phpipam committed:
Small changes

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
merge

phpipam committed:
Changed default table display to 250 and allowed multiple DNS query results

GaryAllan committed:
Bugfix: $this can not be used in static methods

GaryAllan committed:
Bugfix: When scanning subnets don't pollute the generated JSON output with php notice & deprecation errors. Error: Invalid JSON response - JSON_ERROR_SYNTAX.

Ubuntu 18.04 and Fedora 28 generate php72 function deprecated errors in /usr/share/pear/Console/Getopt.php to stdout for PEAR::Ping when debugging is enabled.
Deprecated: The each() function is deprecated. This message will be suppressed on further calls in /usr/share/pear/Console/Getopt.php on line 135"

GaryAllan committed:
Bugfix: When PEAR:Ping scanning hosts don't pollute the generated JSON output with php errors. Error: Invalid JSON response - JSON_ERROR_SYNTAX.

GaryAllan committed:
Bugfix: MAC lookup tool directs to index.php when pretty links disabled. #1951

GaryAllan committed:
Bugfix: All devices, non-admin users gets rack admin-links. #1945

dzhurinskij committed:
Bugfix: Update token_valid_time, max_failures, block_ip validations

GaryAllan committed:
Bugfix: Automatic Discovery not working! #1637
Bugfix: Bug with first_free and custom_field #1849

Integrity constraint violation adding addresses with required custom fields.

GaryAllan committed:
Merge pull request #1961 from adzhurinskij/master

Bugfix: Update token_valid_time, max_failures, block_ip validations

GaryAllan committed:
Enable default caching of bootstrap-custom-light/dark-theme

GaryAllan committed:
Bugfix: get section with space in name gets 404 not found. #1981

urldecode() $_SERVER['REQUEST_URI'] parameters in process_request_uri().

GaryAllan committed:
Bugfix: php72 sizeof() countable fixes

GaryAllan committed:
Bugfix: Fix white background for linked addresses in new UI look. #1939

@korhaldragonir commented on 15 May

When enabling linked addresses the entries had a white background. Just like the text itself.
This made the text unreadable, this small patch fixes that.

GaryAllan committed:
Merge branch 'master' of https://github.com/tsaavik/phpipam into tsaavik-master

GaryAllan committed:
Update PR #1873 for 1.32 changes

GaryAllan committed:
Merge branch 'tsaavik-master'

GaryAllan committed:
Bump phpIPAM version to 1.32 rev0
